### PR TITLE
Update microsoft.identity.client

### PR DESCRIPTION
**Pull request by zohaibhanif**
Issue #28 

- Updated the Microsoft.Identity.Client nuget package to 4.4.0. 
- Also updated the other packages to latest version including Xamarin.Forms, Xamarin.Android, Prism.Forms, Prism.Plugin and Prism.DryIoc.Forms.
- Tested on Android, it is working properly.

### CODE REVIEW

File: shell/ShellApp/Views/MainPage.xaml [hunk 1]
```diff
@@ -9,5 +9,6 @@
            LineBreakMode="WordWrap"
            Text="{Binding JwtPayload}" />
   </ScrollView>
+
```
Comment (dansiegel) [inline shell/ShellApp/Views/MainPage.xaml]: you should not have more than 1 empty line in a row

File: src/AP.AzureADAuth/Services/AuthenticationService.cs [hunk 2]
```diff
@@ -37,8 +37,7 @@ public async Task<AuthenticationResult> LoginAsync()
             if (result is null)
             {
                 var accounts = await _client.GetAccountsAsync();
-                var builder = _client.AcquireTokenInteractive(_configuration.Scopes)
-                                     .WithAccount(accounts.FirstOrDefault());
+                var builder = _client.AcquireTokenInteractive(_configuration.Scopes);
```
Comment (dansiegel) [inline src/AP.AzureADAuth/Services/AuthenticationService.cs]: did MSAL remove the requirement for an account?

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs [hunk 3]
```diff
@@ -51,8 +51,8 @@ public void Destroy()
         {
             _isBusyHelper.Dispose();
             _isBusyHelper = null;
-            LoginCommand.Dispose();
-            LoginCommand = null;
+            //LoginCommand.Dispose();
+            //LoginCommand = null;
```
Comment (dansiegel) [inline src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: I generally do not like including commented out code in PR's unless there is a specific reason we want it... that said why are we commenting this out?

File: src/AP.AzureADAuth/Services/AuthenticationService.cs (hunk 2 above)
Comment (zohaib-avantipoint) [reply src/AP.AzureADAuth/Services/AuthenticationService.cs]: I went through the [MSAL Android consideration](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-xamarin-android-considerations). No account requirement is mentioned over there while acquiring token.

File: shell/ShellApp/Views/MainPage.xaml (hunk 1 above)
Comment (zohaibhanif) [reply shell/ShellApp/Views/MainPage.xaml]: I have done this change in my request next commit.

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs (hunk 3 above)
Comment (zohaibhanif) [reply src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: I have removed this commented lines in my next commit.

**Comment by zohaibhanif** [comment]
@dansiegel  I performed R & D on checks failure issues and found a similar issue reported on MSBuildSdkExtras repo.
https://github.com/onovotny/MSBuildSdkExtras/issues/178

Updating MSBuildSdkExtras  could possibly the solution of it.

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs [hunk 4]
```diff
@@ -41,18 +41,19 @@ public LoginPageViewModel(IAuthenticationService authenticationService, IEventAg
         public async void OnAppearing()
         {
             _logger.TrackEvent("Login Page");
+            var canExecuteCommand = await LoginCommand.CanExecute.FirstAsync();
 
-            await LoginCommand.Execute();
+            if (canExecuteCommand)
```
Comment (dansiegel) [inline src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: there is no reason for this to be a property that you evaluate later...

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs [hunk 5]
```diff
@@ -41,18 +41,19 @@ public LoginPageViewModel(IAuthenticationService authenticationService, IEventAg
...
         }
-
         public void OnDisappearing() { }
 
         public void Destroy()
         {
             _isBusyHelper.Dispose();
             _isBusyHelper = null;
-            LoginCommand.Dispose();
-            LoginCommand = null;
```
Comment (dansiegel) [inline src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: why is this being removed?

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs [hunk 6]
```diff
@@ -41,18 +41,19 @@ public LoginPageViewModel(IAuthenticationService authenticationService, IEventAg
...
             _logger.TrackEvent("Login Page");
+            var canExecuteCommand = await LoginCommand.CanExecute.FirstAsync();
 
-            await LoginCommand.Execute();
+            if (canExecuteCommand)
+            {
+                await LoginCommand.Execute();
+            }
         }
-
```
Comment (dansiegel) [inline src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: you should have a space between methods

File: shell/ShellApp/Views/MainPage.xaml (hunk 1 above)
Comment (dansiegel) [reply shell/ShellApp/Views/MainPage.xaml]: it appears that the multiple empty line has been replaced by a line with whitespace... please remove the whitespace

File: shell/ShellApp.iOS/ShellApp.iOS.csproj [hunk 7]
```diff
@@ -26,8 +26,14 @@
     <WarningLevel>4</WarningLevel>
     <ConsolePause>false</ConsolePause>
     <MtouchArch>x86_64</MtouchArch>
-    <MtouchLink>None</MtouchLink>
+    <MtouchLink>SdkOnly</MtouchLink>
```
[the remark this answers is not in the document]
Comment (dansiegel) [reply shell/ShellApp.iOS/ShellApp.iOS.csproj]: why hasn't this been fixed yet? per our conversation this should have been fixed.

File: src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs (hunk 5 above)
Comment (zohaibhanif) [reply src/AP.AzureADAuth/ViewModels/LoginPageViewModel.cs]: Actually, i was facing an exception"System.ObjectDisposedException: 'Cannot access a disposed object.'" once the LoginCommand gets disposed. It seems framework is trying to access it after its disposal. That's why it commented out these lines and removed them later.

**Pull request merged**